eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/FastaSequenceReader.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.pipeline.io;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.reference.FastaSequenceIndexCreator;
+import htsjdk.samtools.reference.ReferenceSequence;
+import htsjdk.samtools.reference.ReferenceSequenceFile;
+import htsjdk.samtools.reference.ReferenceSequenceFileFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Reads regions from a given FASTA file, and also creates the associated index and dictionary files if they do not
+ * exist.
+ */
+public class FastaSequenceReader {
+
+    private static final Logger logger = LoggerFactory.getLogger(FastaSequenceReader.class);
+
+    private ReferenceSequenceFile fastaSequenceFile;
+
+    private SAMSequenceDictionary sequenceDictionary;
+
+    public FastaSequenceReader(Path fastaPath) throws IOException {
+        fastaSequenceFile = ReferenceSequenceFileFactory.getReferenceSequenceFile(fastaPath, true);
+        sequenceDictionary = fastaSequenceFile.getSequenceDictionary();
+
+        if (sequenceDictionary == null) {
+            logger.info("Sequence dictionary file not found - creating one...");
+            sequenceDictionary = createSequenceDictionary(fastaSequenceFile);
+        }
+        if (!fastaSequenceFile.isIndexed()) {
+            logger.info("Sequence index file not found - creating one...");
+            FastaSequenceIndexCreator.create(fastaPath, true);
+            fastaSequenceFile = ReferenceSequenceFileFactory.getReferenceSequenceFile(fastaPath, true);
+        }
+    }
+
+    /**
+     * Creates a sequence dictionary based on a given FASTA file. Inspired on the Picard tool with the same name.
+     *
+     * @param referenceFile File whose dictionary needs to be created
+     * @return Dictionary associated with the given reference sequence file
+     * @link https://github.com/broadinstitute/picard/blob/master/src/main/java/picard/sam/CreateSequenceDictionary.java
```

Comment:
Curly braces must be used when adding links 
{@link CreateSequenceDictionary.java}

Reply:
Yeah, we should probably use `@see` instead.